### ISSUE TITLE
Consulta sobre "clase padre" al instanciar en multiherencia: ejemplo Avatar

### ISSUE THREAD

**Issue by ssepul21** [IIC2233/Syllabus#160]
(Marcar colocando una `X` entre los corchetes los ítems que ya hiciste, así: `"[X]"`)

* [X] Leí las reglas del foro (https://github.com/IIC2233/Syllabus/wiki/3.-Reglas-del-foro-(issues)).
* [X] Busqué en las *issues* si ya preguntaron mi duda y no encontré nada parecido (https://github.com/IIC2233/syllabus/issues).
* [X] Mi duda no se trata de un tema administrativo o personal, ya que en ese caso debo contactar a mi profe, al ayudante de Bienestar o al correo del curso (coordinacion.iic2233@gmail.com).
* [X] Utilizaré un título descriptivo y llenaré correctamente esta plantilla.
* [X] De ser necesario, colocaré código simple que permita explicar mi problema o duda.

Hola! 
 Mi consulta es algo breve. Pensando en el ejemplo "Avatar" de la clase de multiherencia, **¿cuál es la "clase padre" con la que se inicializa Avatar?** Me da la idea que va a ser MaestroFuego ya que es el primero en ser mencionado dentro de los padres de la clase, pero me gustaría hallar una manera más concreta de poder saberlo.
 Adjunto una captura del código.
![image](https://github.com/IIC2233/Syllabus/assets/142991027/63a4c9d9-013d-4762-826a-3f9f8d7ee395)

Saludos!

**Comment by ssepul21** [comment]
Claro, creo que entiendo un poco mejor. 
 Me queda la duda con el flujo del programa. Lo estaba pensando como por capas, como si fuesen funciones que se llaman una a la otra, formando una especie de secuencia. ¿Es correcto pensarlo de este modo? 
 Algo tipo: ```ClaseABCD``` es inicializado con atributos dados, cuya gran parte pasa a ```ClaseA```, la cual a su vez inicializa la ```ClaseB``` y así hasta llegar a ```object```. Una vez que se llega a el ultimo eslabón, se empiezan a "cerrar" estos objetos, de tal manera que ```ClaseC``` es un objeto ```ClaseD```, pero con atributos extra y así hasta llegar nuevamente a  ```ClaseABCD``` . 
¿Es correcta esta idea? Lo pensaba como un camino de ida y vuelta.

Espero se haya entendido la idea

**Comment by mskdancers** [comment]
Claro, es una buena manera de pensarlo. En efecto, son funciones (métodos, más específicamente) anidadas, por lo que se llaman hacia arriba y luego se van "cerrando" como dices tú. Si quieres visualizarlo mejor puedes poner _prints_ dentro de los inicializadores y ver cuándo está entrando a qué clase 👀, te dejo el ejemplo acá abajo. Saludos!!

```python
class ClaseA:
    def __init__(self, a, *args, **kwargs):
        print("Inicializando clase A...")
        super().__init__(*args, **kwargs)
        self.a = a
        print("Inicialización de clase A lista!")

class ClaseB:
    def __init__(self, b, *args, **kwargs):
        print("Inicializando clase B...")
        super().__init__(*args, **kwargs)
        self.b = b
        print("Inicialización de clase B lista!")

class ClaseC:
    def __init__(self, c, *args, **kwargs):
        print("Inicializando clase C...")
        super().__init__(*args, **kwargs)
        self.c = c
        print("Inicialización de clase C lista!")

class ClaseD:
    def __init__(self, d, *args, **kwargs):
        print("Inicializando clase D...")
        super().__init__(*args, **kwargs)
        self.d = d
        print("Inicialización de clase D lista!")

class ClaseABCD(ClaseA, ClaseB, ClaseC, ClaseD):
    def __init__(self, *args, **kwargs):
        print("Inicializando clase ABCD...")
        super().__init__(*args, **kwargs)
        print("Inicialización de clase ABCD lista!")

abcd = ClaseABCD("a", "b", "c", "d")
```
